content/reference/wadm/model.md

```diff
@@ -1,8 +1,111 @@
+
+## Traits
+Traits are, as their name applies, metadata associated with a `component`. The OAM trait system is completely extensible, so as wadm gains more functionality, it can support more traits. Right now, the following traits are supported:
+
+* `spreadscaler`
+* `linkdef`
+
+### Spread Scaler
+The `spreadscaler` trait contains a specification for how you would like to scale a set number of instances of an actor. We call it a _spread_ scaler because you declare how you would like the instances of that actor spread across the hosts within your lattice by specifying targets with host _labels_. You can think of this like affinity and anti-affinity rules combined with a scale specification.
+
+Take a look at the following sample `spreadscaler` spec:
+
+```yaml
+      traits:
+        - type: spreadscaler
+          properties:
+            replicas: 4
+            spread:
+              - name: eastcoast
+                requirements:
+                  zone: us-east-1
+                weight: 80
+              - name: westcoast
+                requirements:
+                  zone: us-west-1
+                weight: 20
+```
+
+This definition states that, for this component (a spread scaler can apply to an `actor` or `capability`), you want a total of 4 instances, with 80% of them going to hosts with the `zone` label set to `us-east-1` and 20% of them going to hosts with the `zone` label set to `us-west-1`. Because this system uses labels as selectors, and you can set any arbitrary label on your hosts, you can define practically any conditions for the spread rules.
+
+⚠️ _NOTE_: if you define a label/value pair requirement and wadm is unable to find hosts that match this constraint, it will consider this a deployment failure and will _not_ fall back to arbitrary placement.
+
+### Link Definition
+The `linkdef` trait links two components together with a set of configuration values.
+
+```yaml
+- type: linkdef
+    properties:
+    target: webcap
+    values:
+        port: 8080
+```
+
+Quite possibly one of the best features of specifying link definitions in a wadm file as opposed to using imperative `wash` commands is that you do _not need to use the source or target's public key_. If you've used `wash` to specify link definitions before, you know the syntax can be verbose.
+
+The value of the `target` field is a _component_ whose `name` field matches that. The `values` is a simple key-value map that will be passed as link definition configuration data at deployment time.
+
+⚠️ _NOTE_: while wadm can "claim" resources like actors and providers and differentiate between wadm-managed versus unmanaged, this is not so with link definitions. If you manually push new link definition information that overrides or conflicts with the link definitions in your wadm spec, you could experience unexpected behavior until wadm corrects for it. You must take extreme care that wadm-managed link definitions don't conflict with external link definitions.
```

Comment:
I think it would be useful to have a "putting it all together" section to show a full, real, application spec like for kvcounter https://github.com/wasmCloud/examples/blob/main/actor/kvcounter/wadm.yaml

Reply:
Instead I think I'll just link to that full file